contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1156,3 +1165,391 @@ host_name(PG_FUNCTION_ARGS)
+				 errmsg("Expected 1 statement but get %d statements after parsing",
+						list_length(res))));
+
+	stmt = parsetree_nth_stmt(res, 0);
+	GrantRoleStmt *grant_role = (GrantRoleStmt *) stmt;
+	granted = (AccessPriv *) linitial(grant_role->granted_roles);
+
+	/* This is ALTER ROLE statement */
+	grantee = (RoleSpec *) linitial(grant_role->grantee_roles);
+
+	/* Rewrite granted and grantee roles */
+	pfree(granted->priv_name);
+	granted->priv_name = user;
+
+	pfree(grantee->rolename);
+	grantee->rolename = member;
+
+	rewrite_object_refs(stmt);
+
+	return res;
+}
+
+Datum sp_droprolemember(PG_FUNCTION_ARGS)
+{
+	List *parsetree_list;
+	ListCell *parsetree_item;
+	const char *prev_current_user;
+	const char *saved_dialect = GetConfigOption("babelfishpg_tsql.sql_dialect", true, true);
+
+	prev_current_user = GetUserNameFromId(GetUserId(), false);
+	bbf_set_current_user("sysadmin");
+
+	PG_TRY();
+	{
+		set_config_option("babelfishpg_tsql.sql_dialect", "tsql",
+							(superuser() ? PGC_SUSET : PGC_USERSET),
+							PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+
+		char *rolname = text_to_cstring(PG_GETARG_TEXT_PP(0));
+		char *membername = text_to_cstring(PG_GETARG_TEXT_PP(1));
+
+		/* Advance cmd counter to make the delete visible */
+		CommandCounterIncrement();
+
+		parsetree_list = gen_sp_droprolemember_subcmds(rolname, membername);
+
+		/* Run all subcommands */
+		foreach(parsetree_item, parsetree_list)
+		{
+			Node *stmt = ((RawStmt *) lfirst(parsetree_item))->stmt;
+			PlannedStmt *wrapper;
+
+			/* need to make a wrapper PlannedStmt */
+			wrapper = makeNode(PlannedStmt);
+			wrapper->commandType = CMD_UTILITY;
+			wrapper->canSetTag = false;
+			wrapper->utilityStmt = stmt;
+			wrapper->stmt_location = 0;
+			wrapper->stmt_len = 16;
+
+			/* do this step */
+			ProcessUtility(wrapper,
+				"(ALTER ROLE )",
+				false,
+				PROCESS_UTILITY_SUBCOMMAND,
+				NULL,
+				NULL,
+				None_Receiver,
+				NULL);
+
+			/* make sure later steps can see the object created here */
+			CommandCounterIncrement();
+		}
+	}
+	PG_CATCH();
+	{
+		/* Clean up. Restore previous state. */
+		bbf_set_current_user(prev_current_user);
+		set_config_option("babelfishpg_tsql.sql_dialect", saved_dialect,
+							(superuser() ? PGC_SUSET : PGC_USERSET),
+							PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+		PG_RE_THROW();
+	}
+	PG_END_TRY();
+}
+
+static List *
+gen_sp_droprolemember_subcmds(const char *user, const char *member)
+{
+	StringInfoData query;
+	List *res;
+	Node *stmt;
+	AccessPriv *granted;
+	RoleSpec *grantee;
+
+	initStringInfo(&query);
+	appendStringInfo(&query, "ALTER ROLE dummy DROP MEMBER dummy; ");
+	res = raw_parser(query.data, RAW_PARSE_DEFAULT);
+
+	if (list_length(res) != 1)
+		ereport(ERROR,
+				(errcode(ERRCODE_SYNTAX_ERROR),
+				 errmsg("Expected 1 statement but get %d statements after parsing",
+						list_length(res))));
+
+	stmt = parsetree_nth_stmt(res, 0);
+	GrantRoleStmt *grant_role = (GrantRoleStmt *) stmt;
```

Comment:
Please put variable declaration in front of other part of code.

Reply:
Fixed it in next revision by placing variable declaration at front.